### ISSUE TITLE
Full app reload, always

### ISSUE THREAD

**Issue by jods4** [hannoeru/vite-plugin-pages#55]
While testing 0.11.4, I noticed that no matter what I change in a page, a full page reload is triggered.
For example, if I change a button label -> full page reload.

I suspect it was introduced by the fix for #52.
Page change -> potential `<route>` block change -> regenerate routing (always?) -> HMR not possible for `routes.js` -> full reload (?)

If `<route>` didn't change, that process should stop early and not induce an HMR failure.

**Comment by hannoeru** [comment]
I can't repro it. Can you provide more context or reproduction?

**Comment by jods4** [comment]
@hannoeru sorry about that. 
It took me a bit of time to manage to build a repro because this bug is triggered by something very specific to my project.

Here's a repro: https://github.com/jods4/bug-vite-pages
Click on the counter button, then save `index.vue` (no change) and it triggers a full reload.

I found what specifically triggers the bug:
I have a `extendRoute` option that removes some properties that were created by `<route>` (in real life it does a little more, but for this repro, `delete route.nav` is enough.
https://github.com/jods4/bug-vite-pages/blob/master/vite.config.js#L14

Armed with this knowledge, I dug into the codebase and I think I found an explanation.
The plugin performs a full reload when `isRouteBlockChanged`:
https://github.com/hannoeru/vite-plugin-pages/blob/6096a331ba5bd60283da28f0cb20e50d1e2125b5/src/generate.ts#L145-L158

This function doesn't actually compare the `<route>` block text to see if it has changed.
Instead it compares the old `route` with the same route with the new block applied.
This doesn't account for the fact that `extendRoute` may have been used to modify properties that initially came from the route block!

I suppose there are several ways to fix this, e.g.:
- Call `extendRoute` on the new object before doing the comparison;
- Keep (and compare) the _text_ of the route block.
